### PR TITLE
feat(discover2): Enable browser right-click functions on SavedQuery cards

### PR DESCRIPTION
**Pull request by leedongwei**
## Before
image

## After
image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/querycard.tsx [hunk 1]
```diff
@@ -56,7 +61,7 @@ class QueryCard extends React.PureComponent<Props> {
   }
 }
 
-const StyledQueryCard = styled('button')`
+const StyledQueryCard = styled(Link)`
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/querycard.tsx]: Will this trigger DOM nesting warnings when the context menu is opened?

Comment (leedongwei) [reply src/sentry/static/sentry/app/views/eventsV2/querycard.tsx]: Nope, the parent elements are `div` all the way up. I've checked in Chrome and FF.

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Comment by leedongwei** [comment]
Rebased against `origin/master` and force-pushed because Percy was lighting up on every page. 

I had modified `link.jsx` which is used all over Sentry and I wanted to be sure it wasn't me.

**Review by markstory — APPROVED** [review]

**Pull request merged**